sdk/program/src/bignum.rs

```diff
@@ -0,0 +1,851 @@
+//! Solana BigNumber
+//!
+//! Rust-based BigNumber provides safe math for arbitrarily large numbers.
+//! Primarily focues on BPF program usage, non-BPF is also supported by
+//! backing from the OpenSSL implementation
+use std::fmt;
+
+#[repr(C)]
+#[derive(Debug)]
+/// BPF structure for sending/receiving BigNumber data to/from syscalls
+pub struct FfiBigNumber {
+    pub data_len: u64,     // Length of bytes of data file
+    pub is_negative: bool, // False (0) or True (1) if data is positive or negative
+    pub data: u64, // Pointer to the variable length big endian array represneting the vallue of BigNumber
```

Comment:
nit: called "data" here but 'value" down below

Reply:
Hmmm... are you suggesting that the property in FfiBigNumber and the one in BigNumber should be named the same?

Reply:
Yeah, did you have a reason not to?

Reply:
@jackcmay No, but I'm fully getting into the rigor of crossing many T's :) I will make the change to have consistency

Reply:
Cool, and pull as much out of the unsafe blocks as possible :-)